naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/resolvers/NaptimeResolver.scala

```diff
@@ -268,4 +296,10 @@ class NaptimeResolver extends DeferredResolver[SangriaGraphQlContext] with Stric
   }
 
 
+  private[this] def getInternalRequestContext(context: SangriaGraphQlContext): SangriaGraphQlContext = {
+    val originalRequestHeader = context.requestHeader
+    val newHeaders = context.requestHeader.headers.remove("X-Coursera-User").add("X-Coursera-Source" -> "internal")
```

Comment:
is it possible to move this to assembler's RemoteFetcher? that way implementation detail of how coursera handles auth isn't leaked onto the Naptime project and I believe that is the place where we have the final headers and instantiate the client so it feels like a more natural place to do this conditional logic.

Reply:
yeah I think that's doable and that all sounds reasonable. let me see